### PR TITLE
Feature/mds-852 Mines tailings file picker

### PR DESCRIPTION
**Pull request by james-hollinger**
Create a new MinesTailingsFilePicker component that handles the redux and API calls for the FilePicker in the EditTailingsReportModal. Replaces the FileUpload component in EditTailingsReportForm with this new component.

Also fixes some linting errors in EditTailingsReportForm and EditTailingsReportModal.

### CODE REVIEW

File: frontend/src/components/Forms/EditTailingsReportForm.js [hunk 1]
```diff
@@ -4,20 +4,26 @@ import { Field, reduxForm } from "redux-form";
...
 
 const propTypes = {
   handleSubmit: PropTypes.func.isRequired,
   closeModal: PropTypes.func.isRequired,
   title: PropTypes.string.isRequired,
-  statusOptions: PropTypes.array.isRequired,
-  selectedDocument: PropTypes.object,
+  statusOptions: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.string,
```
Comment (JustinMacaulay) [inline frontend/src/components/Forms/EditTailingsReportForm.js]: I believe there is a CustomPropType for this already.

File: frontend/src/components/modalContent/EditTailingsReportModal.js [hunk 2]
```diff
@@ -5,13 +5,20 @@ import EditTailingsReportForm from "@/components/Forms/EditTailingsReportForm";
 const propTypes = {
   onSubmit: PropTypes.func.isRequired,
   title: PropTypes.string.isRequired,
-  statusOptions: PropTypes.array.isRequired,
-  initialValues: PropTypes.object,
-  selectedDocument: PropTypes.object,
+  initialValues: PropTypes.objectOf(PropTypes.any),
+  statusOptions: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.string,
```
Comment (JustinMacaulay) [inline frontend/src/components/modalContent/EditTailingsReportModal.js]: Same as above, in common.js there is a CustomPropType that handles this.

File: frontend/src/components/modalContent/EditTailingsReportModal.js [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
 
 const propTypes = {
   onSubmit: PropTypes.func.isRequired,
   title: PropTypes.string.isRequired,
-  statusOptions: PropTypes.array.isRequired,
-  initialValues: PropTypes.object,
-  selectedDocument: PropTypes.object,
+  initialValues: PropTypes.objectOf(PropTypes.any),
+  statusOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+  selectedDocument: PropTypes.shape({
```
Comment (nayr974) [inline frontend/src/components/modalContent/EditTailingsReportModal.js]: And here :)

File: frontend/src/reducers/mineReducer.js [hunk 4]
```diff
@@ -43,6 +44,11 @@ const mineReducer = (state = initialState, action) => {
         ...state,
         mineBasicInfoList: action.payload,
       };
+    case actionTypes.STORE_MINE_DOCUMENTS:
+      return {
+        ...state,
+        mineDocuments: action.payload.mine_documents,
```
Comment (nayr974) [inline frontend/src/reducers/mineReducer.js]: Curious if we're now duplicating documents in our store, once as child properties of the tailings reports and here? Should we refactor the tailings screen after this change to map based on a relationship, or is this ok?

File: frontend/src/components/Forms/EditTailingsReportForm.js [hunk 5]
```diff
@@ -1,23 +1,25 @@
...
-import { DOCUMENT, EXCEL } from "@/constants/fileTypes";
 
 const propTypes = {
   handleSubmit: PropTypes.func.isRequired,
   closeModal: PropTypes.func.isRequired,
   title: PropTypes.string.isRequired,
-  statusOptions: PropTypes.array.isRequired,
-  selectedDocument: PropTypes.object,
+  statusOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+  selectedDocument: PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (JustinMacaulay) [reply frontend/src/components/Forms/EditTailingsReportForm.js]: There is a custom prop type for this in my PR #401.

[the remark this answers is not in the document]
Comment (james-hollinger) [reply frontend/src/components/Forms/EditTailingsReportForm.js]: Implemented

File: frontend/src/components/mine/Tailings/MineTailingsFilePicker.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  fetchMineDocuments,
+  addMineDocumentToExpectedDocument,
+} from "@/actionCreators/mineActionCreator";
+import { UPLOAD_MINE_EXPECTED_DOCUMENT_FILE } from "@/constants/API";
+import { createDropDownList } from "@/utils/helpers";
+import { DOCUMENT, EXCEL } from "@/constants/fileTypes";
+import FilePicker from "@/components/common/FilePicker";
+
+const propTypes = {
+  selectedDocument: PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (james-hollinger) [reply frontend/src/components/mine/Tailings/MineTailingsFilePicker.js]: Implemented

File: frontend/src/components/modalContent/EditTailingsReportModal.js (hunk 3 above)
Comment (james-hollinger) [reply frontend/src/components/modalContent/EditTailingsReportModal.js]: Implemented

File: frontend/src/components/mine/Tailings/MineTailingsFilePicker.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { getMineDocuments } from "@/selectors/mineSelectors";
+import {
+  fetchMineDocuments,
+  addMineDocumentToExpectedDocument,
+} from "@/actionCreators/mineActionCreator";
+import { UPLOAD_MINE_EXPECTED_DOCUMENT_FILE } from "@/constants/API";
+import { createDropDownList } from "@/utils/helpers";
+import { DOCUMENT, EXCEL } from "@/constants/fileTypes";
```
[the remark this answers is not in the document]
Comment (james-hollinger) [reply frontend/src/components/mine/Tailings/MineTailingsFilePicker.js]: 👍

**Comment by james-hollinger** [comment]
Tests have been added. Some integration work to sync up this component with Justin's attached file list component needs to be done, which will be in another PR.

**Pull request merged**